### PR TITLE
Trying to get the git describe build string to include the python version

### PR DESCRIPTION
**Pull request by asmeurer**
This doesn't work because this information is not known when the templating is
run.

### CODE REVIEW

**Comment by dan-blanchard** [comment]
I've been running into this exact same issue. I'm glad to see you guys are working on it.

File: conda_build/environ.py [hunk 1]
```diff
@@ -84,6 +88,9 @@ def get_dict(m=None, prefix=None):
     d['NPY_VER'] = get_npy_ver()
     d['SRC_DIR'] = source.get_dir()
 
+    if os.path.isdir(os.path.join(d['SRC_DIR'], '.git')):
+        d.update(**get_git_build_info(m, d['SRC_DIR']))
```
Comment (asmeurer) [inline conda_build/environ.py]: I guess this was an incorrect merge conflict resolution.

**Comment by asmeurer** [comment]
This doesn't work.

I give up trying to fix the code flow to make it work. The Jinja templating needs to have access to the rest ot the Metadata object metadata to correctly add it to the GIT_BUILD_STR, but when it is run it doesn't (and can't, because that information can't even be read until after the meta.yaml is templated). 

The jinja2 templating is broken by design, and I honestly think we should just get rid of it, and replace it with some kind of variable system where the meta.yaml can be parsed and the variables replaced at a later time (possibly many times as more information is known throughout the build process).

**Comment by ilanschnell** [comment]
I agree, we should get rid of the jinja2 templating.  It has been a source of confusion for a long time.  Maybe we can have special variables (such as `__GIT_BUILD_STR__`) in the meta.yaml file.

**Comment by dan-blanchard** [comment]
Not that anybody asked for my opinion, but I completely agree.  The jinja2 system is really confusing for beginners, as jinja2 isn't even a requirement for conda-build.  I will say that if you do this though, it'd be nice if there was a special variable for the current date (formatted as a version like `2015.03.10`).  That's the only thing I've ever used the templating for other than `GIT_BUILD_STR`.

**Pull request closed without merging**